### DISCUSSION TITLE
C# Feature Request: Infinite loop syntax without head or tail condition

### DISCUSSION THREAD

**Discussion by longwolf** [dotnet/csharplang#4580] [Language Ideas] [answered]
Please enable
`do {`
`    // body`
`};`
without a dedicated exit condition.

Background: I just stumbled upon [this discussion about a new "loop" keyword for infinite / break controlled loops](https://developercommunity.visualstudio.com/t/Infinite-Loop/383099).

Because I couldn't (and wouldn't) answer there (discussion closed and suggested to be moved here), I thought I'd put the most obvious suggestion without new keywords here.

**Comment by CyrusNajmabadi** [comment]
`while(true)` already exists and is already idiomatic. :-)

**Comment by YairHalberstadt** [comment, marked as answer]
See earlier discussion at https://github.com/dotnet/csharplang/issues/2475
